bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeAirDevice.java

```diff
@@ -248,37 +233,40 @@ public void setDeviceLight(DatagramSocket clientSocket, int value) throws GreeEx
     /**
      * @param value set temperature in degrees Celsius or Fahrenheit
      */
-    public void setDeviceTempSet(DatagramSocket clientSocket, int value) throws GreeException {
-        int newVal = value;
-        int outVal = value;
-        // Get Celsius or Fahrenheit from status message
-        Integer CorF = getIntStatusVal(GREE_PROP_TEMPUNIT);
-        // TODO put a param in openhab to allow setting this from the config
-
+    public void setDeviceTempSet(DatagramSocket clientSocket, QuantityType<?> temp) throws GreeException {
         // If commanding Fahrenheit set halfStep to 1 or 0 to tell the A/C which F integer
         // temperature to use as celsius alone is ambigious
-        int halfStep = 0; // default to C
-
-        int[] retList = validateTemperatureRangeForTempSet(newVal, CorF);
-        newVal = retList[0];
-        CorF = retList[1];
+        Double newVal = temp.doubleValue();
+        int CorF = temp.getUnit() == SIUnits.CELSIUS ? TEMP_UNIT_CELSIUS : TEMP_UNIT_FAHRENHEIT; // 0=Celsius,
+                                                                                                 // 1=Fahrenheit
+        if (((CorF == TEMP_UNIT_CELSIUS) && (newVal < TEMP_MIN_C || newVal > TEMP_MAX_C))
+                || ((CorF == TEMP_UNIT_FAHRENHEIT) && (newVal < TEMP_MIN_F || newVal > TEMP_MAX_F))) {
+            throw new IllegalArgumentException("Temp Value out of Range");
```

Comment:
You could convert the QuantityType with `toUnit()` to Celsius or Fahrenheit. Then, you need to check the range only once.

Reply:
this is more code, more overhead and feels for me a little bit strange. I send an out-of-range value through the converter and then check that it's out of range?

Reply:
@fwolter how do we proceed on this?

Reply:
It's a matter of taste. Your argument is valid. IMHO it's ok.

Reply:
ok